lib/piwik/index.js

```diff
@@ -64,8 +67,35 @@ Piwik.prototype.track = function(track){
   var name = track.proxy('properties.name') || track.proxy('properties.label');
   var value = track.value() || track.revenue();
 
+  var options = track.options('Piwik');
+
+  // Add this if we enable setting your own customVariableLimit
+  //var customVariableLimit = this.options.customVariableLimit;
+
+  // Not currently enabling setting your own variable limit, setting this to the default
+  var customVariableLimit = 5;
+
+  var customVariables = options.customVars || options.cvar;
+  var formattedVars = {};
+
+  if (is.array(customVariables)) {
+    formattedVars = reduce(customVariables, function(formattedVars, value, i){
+      if (i <= customVariableLimit) formattedVars[i] = value;
+      return formattedVars;
+    }, {});
+  }
+
+  if (is.object(customVariables)) formattedVars = customVariables;
```

Comment:
It won't send anything if you provide them with the wrong type, there are a couple of tests in there for incorrect types. 

No, `is.object` returns false for arrays.